contrib/src/databases.rs

```diff
@@ -0,0 +1,85 @@
+pub extern crate r2d2;
+
+use std::marker::{Send, Sized};
+use std::io;
+use std::error::Error;
+use rocket::config;
+use rocket::Rocket;
+use rocket::config::ConfigError;
+use rocket::fairing::{Fairing, Info, Kind};
+
+use self::r2d2::ManageConnection;
+
+#[cfg(feature = "diesel_sqlite")]
+pub extern crate diesel;
+
+#[cfg(feature = "pg_pool")]
+pub extern crate postgres;
+#[cfg(feature = "pg_pool")]
+pub extern crate r2d2_postgres;
+
+pub struct DatabaseConfig {
+    pub url: String,
+    pub pool_size: u32,
+}
+
+pub fn database_config(name: &str, from: &config::Config) -> Option<DatabaseConfig> {
```

Comment:
This is super hard to read. Don't be afraid to use `?` to make this easier to follow.

Also, why not store the `&str` in the `DatabaseConfig`? Something like:

```rust
struct DatabaseConfig<'a> {
    pub url: &'a str,
    pub pool_size: u32,
}